gateway/src/apicast/configuration_loader.lua

```diff
@@ -219,6 +219,8 @@ function lazy.rewrite(configuration, host)
     ngx.log(ngx.WARN, 'failed to acquire lock to lazy load: ', host, ' error: ', err)
   end
 
+  collectgarbage()
```

Comment:
In some configurations when reloading, this line is not executed. I think we need to call `collectgarbage` also in `refresh_configuration`, or somewhere else that runs when re-loading in 'boot' mode.

Reply:
Pushed a commit fixing this.